crates/cubecl-runtime/src/feature_set.rs

```diff
@@ -0,0 +1,41 @@
+use crate::memory_management::MemoryDeviceProperties;
+use std::collections::BTreeSet;
+
+/// The set of [features](Feature) supported by a [runtime](Runtime).
+#[derive(Debug)]
+pub struct ClientProperties<Feature: Ord + Copy> {
```

Comment:
I don't think the name is right, yes it's accessed by the client, but isn't it better to name it `DeviceProperties` instead?

Reply:
Yeah makes sense, fixed!